### ISSUE TITLE
build: pie builds are broken

### ISSUE THREAD

**Issue by bassam** [rook/rook#204]
`build/run make -j4 STATIC=0 PIE=1` fails with permission denied.

**Comment by bassamtabbara** [comment]
After #206 PIE seems to be partially working. The new issue seems to be libblkid not supporting -fPIC:

```
# github.com/rook/rook/cmd/rookd
/usr/local/go/pkg/tool/linux_amd64/link: running aarch64-linux-gnu-g++ failed: exit status 1
/usr/lib/gcc-cross/aarch64-linux-gnu/5/../../../../aarch64-linux-gnu/bin/ld: /usr/lib/aarch64-linux-gnu/libblkid.a(libblkid_la-ntfs.o): relocation R_AARCH64_ADR_PREL_PG_HI21 against external symbol `stderr@@GLIBC_2.17' can not be used when making a shared object; recompile with -fPIC
/usr/lib/gcc-cross/aarch64-linux-gnu/5/../../../../aarch64-linux-gnu/bin/ld: /usr/lib/aarch64-linux-gnu/libblkid.a(libblkid_la-ntfs.o)(.text+0x29c): unresolvable R_AARCH64_ADR_PREL_PG_HI21 relocation against symbol `stderr@@GLIBC_2.17'
/usr/lib/gcc-cross/aarch64-linux-gnu/5/../../../../aarch64-linux-gnu/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
```
